Upgrade to Pydantic 2.x

akx opened the pull request:
Resolves that TODO about doing this.

akx committed:
Upgrade to Pydantic 2.x

rambo reviewed (changes requested):
Needs a version bump, use `bump2version` I guess `major` since we change major dep version.

akx committed:
`uvx bump2version major`

akx commented:
> Needs a version bump, use `bump2version` I guess `major` since we change major dep version.

Done, though feels like this should be done in a release PR based on the changes since the last version (e.g. conventional-commits/semantic-release does this by way of [an exclamation mark](https://stackoverflow.com/q/70781241/51685) in the commit-type word).

rambo commented:
I agree that we should start using conventional commits but that requires everyone to make that effort. Perhaps next Hackathon should have a workshop on it and then we can add CI check that rejects commits that do not follow the rules to repos.

rambo reviewed (approved):
:shipit:
